endpoint/main.py

```diff
@@ -0,0 +1,131 @@
+import os
+import logging
+import logging.config
+import json
+import time
+from urllib.parse import urlparse
+import uvicorn
+
+import redis
+import jwt
+import boto3
+from botocore.exceptions import ClientError
+
+from typing import Optional
+
+from fastapi import FastAPI, Request
+from fastapi.responses import JSONResponse
+from pydantic import BaseModel
+
+
+logger = logging.getLogger(__name__)
+
+logging.config.dictConfig({
+    'version': 1,
+    'disable_existing_loggers': False,
+    'formatters': {
+        'simple': {
+            'format': '%(levelname)s: %(message)s',
+        },
+        'verbose': {
+            'format': '[%(asctime)s][%(module)s] %(levelname)s: %(message)s',
+        }
+    },
+    'handlers': {
+        'console': {
+            'level': 'DEBUG',
+            'class': 'logging.StreamHandler',
+            'formatter': 'verbose',
+        }
+    },
+    'loggers': {
+        '': {
+            'handlers': ['console'],
+            'level': 'DEBUG',
+            'propagate': False,
+        }
+    }
+})
+
+client = redis.Redis(
+    host=os.environ['REDIS_HOST'],
+    port=int(os.environ.get('REDIS_PORT', '6379')),
+    charset="utf-8",
+    decode_responses=True,
+)
+
+register_queue = os.environ['REDIS_REGISTER_QUEUE_KEY']
+progress_set = os.environ['REDIS_REGISTER_PROGRESS_KEY']
+success_set = os.environ['REDIS_REGISTER_SUCCESS_KEY']
+failure_set = os.environ['REDIS_REGISTER_FAILURE_KEY']
+wait_time = float(os.environ.get('WAIT_TIME'))
```

Comment:
There is no default value passed to the `.get()` method. So when no "WAIT_TIME" env var is present, this line will fail, because it will be essentially `float(None)`.

Something like the following is better:

```python
wait_time = float(os.environ.get('WAIT_TIME', '0.1'))
```